File: tags/last-successful/jWebSocket/jWebSocket-1.0-2014-12-03_23-02-53/shared/J2SE/jWebSocketCommon/org/jwebsocket/kit/WebSocketSubProtocol.java
//	---------------------------------------------------------------------------
//	jWebSocket - WebSocketSubProtocol (Community Edition, CE)
//	---------------------------------------------------------------------------
//	Copyright 2010-2014 Innotrade GmbH (jWebSocket.org)
//	Alexander Schulze, Germany (NRW)
//
//	Licensed under the Apache License, Version 2.0 (the "License");
//	you may not use this file except in compliance with the License.
//	You may obtain a copy of the License at
//
//	http://www.apache.org/licenses/LICENSE-2.0
//
//	Unless required by applicable law or agreed to in writing, software
//	distributed under the License is distributed on an "AS IS" BASIS,
//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//	See the License for the specific language governing permissions and
//	limitations under the License.
//	---------------------------------------------------------------------------
package org.jwebsocket.kit;

import org.jwebsocket.config.JWebSocketCommonConstants;

/**
 *
 * @author Alexander Schulze
 */
public class WebSocketSubProtocol {

	private String mSubProt = null;
	private String mNameSpace = null;
	private String mFormat = null;
	private WebSocketEncoding mEncoding;

	/**
	 *
	 * @param aSubProt
	 * @param aEncoding
	 */
	public WebSocketSubProtocol(String aSubProt, WebSocketEncoding aEncoding) {
		this.mSubProt = aSubProt;
		this.mEncoding = aEncoding;

		if (JWebSocketCommonConstants.WS_SUBPROT_JSON.equals(aSubProt)) {
			mNameSpace = JWebSocketCommonConstants.WS_SUBPROT_PREFIX;
			mFormat = JWebSocketCommonConstants.WS_FORMAT_JSON;
		} else if (JWebSocketCommonConstants.WS_SUBPROT_XML.equals(aSubProt)) {
			mNameSpace = JWebSocketCommonConstants.WS_SUBPROT_PREFIX;
			mFormat = JWebSocketCommonConstants.WS_FORMAT_XML;
		} else if (JWebSocketCommonConstants.WS_SUBPROT_CSV.equals(aSubProt)) {
			mNameSpace = JWebSocketCommonConstants.WS_SUBPROT_PREFIX;
			mFormat = JWebSocketCommonConstants.WS_FORMAT_CSV;
		} else if (JWebSocketCommonConstants.WS_SUBPROT_TEXT.equals(aSubProt)) {
			mNameSpace = JWebSocketCommonConstants.WS_SUBPROT_PREFIX;
			mFormat = JWebSocketCommonConstants.WS_FORMAT_TEXT;
		} else if (JWebSocketCommonConstants.WS_SUBPROT_BINARY.equals(aSubProt)) {
			mNameSpace = JWebSocketCommonConstants.WS_SUBPROT_PREFIX;
			mFormat = JWebSocketCommonConstants.WS_FORMAT_BINARY;
		}
	}

	@Override
	public int hashCode() {
		return mSubProt.hashCode() + mEncoding.getEncoding();
	}

	@Override
	public boolean equals(Object aObj) {
		if (aObj instanceof WebSocketSubProtocol) {
			WebSocketSubProtocol lOther = (WebSocketSubProtocol) aObj;
			return mSubProt.equals(lOther.mSubProt) && mEncoding.equals(lOther.mEncoding);
		} else {
			return super.equals(aObj);
		}
	}

	@Override
	public String toString() {
		StringBuilder lBuff = new StringBuilder();
		lBuff.append(mSubProt).append('[').append(mEncoding.name()).append("]");
		return lBuff.toString();
	}

	/**
	 * @return the mSubProt
	 */
	public String getSubProt() {
		return mSubProt;
	}

	/**
	 * @return the mEncoding
	 */
	public WebSocketEncoding getEncoding() {
		return mEncoding;
	}

	/**
	 * @return the namespace of the sub protocol
	 */
	public String getNameSpace() {
		return mNameSpace;
	}

	/**
	 * @return the format of the sub protocol
	 */
	public String getFormat() {
		return mFormat;
	}
}

File: tags/last-successful/jWebSocket/jWebSocket-1.0-2015-03-03_22-55-24/Windows Phone API/Common Code/org/jwebsocket/client/kit/WebSocketSubProtocol.cs
//	---------------------------------------------------------------------------
//	jWebSocket - TimeoutOutputStreamNIOWriter
//	Copyright (c) 2011 Alexander Schulze, Innotrade GmbH
//	---------------------------------------------------------------------------
//	This program is free software; you can redistribute it and/or modify it
//	under the terms of the GNU Lesser General Public License as published by the
//	Free Software Foundation; either version 3 of the License, or (at your
//	option) any later version.
//	This program is distributed in the hope that it will be useful, but WITHOUT
//	ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
//	FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for
//	more details.
//	You should have received a copy of the GNU Lesser General Public License along
//	with this program; if not, see <http://www.gnu.org/licenses/lgpl.html>.
//	---------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ClientLibrary.org.jwebsocket.client.config;

namespace ClientLibrary.org.jwebsocket.client.kit
{
    /// <summary>
    /// Author Rolando Betancourt Toucet
    /// </summary>
    public class WebSocketSubProtocol
    {
        private string mSubProt;
        private string mNameSpace;
        private string mFormat;
        private WebSocketEncoding mEncoding;

        public WebSocketSubProtocol(string aSubProt, WebSocketEncoding aEncoding)
        {
            this.mSubProt = aSubProt;
            this.mEncoding = aEncoding;

            if (WebSocketConstants.WS_SUBPROT_JSON.Equals(aSubProt))
            {
                mNameSpace = WebSocketConstants.WS_SUBPROT_PREFIX;
                mFormat = WebSocketConstants.WS_FORMAT_JSON;
            }
            else if (WebSocketConstants.WS_SUBPROT_XML.Equals(aSubProt))
            {
                mNameSpace = WebSocketConstants.WS_SUBPROT_PREFIX;
                mFormat = WebSocketConstants.WS_FORMAT_XML;
            }
            else if (WebSocketConstants.WS_SUBPROT_CSV.Equals(aSubProt))
            {
                mNameSpace = WebSocketConstants.WS_SUBPROT_PREFIX;
                mFormat = WebSocketConstants.WS_FORMAT_CSV;
            }
            else if (WebSocketConstants.WS_SUBPROT_TEXT.Equals(aSubProt))
            {
                mNameSpace = WebSocketConstants.WS_SUBPROT_PREFIX;
                mFormat = WebSocketConstants.WS_FORMAT_TEXT;
            }
            else if (WebSocketConstants.WS_SUBPROT_BINARY.Equals(aSubProt))
            {
                mNameSpace = WebSocketConstants.WS_SUBPROT_PREFIX;
                mFormat = WebSocketConstants.WS_FORMAT_BINARY;
            }
        }

        public override int GetHashCode()
        {
            return mSubProt.GetHashCode() + (int)mEncoding;
        }

        public override bool Equals(object aObj)
        {
            if (aObj is WebSocketSubProtocol)
            {
                WebSocketSubProtocol lOther = (WebSocketSubProtocol)aObj;
                return mSubProt.Equals(lOther.mSubProt) && mEncoding.Equals(lOther.mEncoding);
            }
            else
                return base.Equals(aObj);
        }

        public override string ToString()
        {
            StringBuilder lBuff = new StringBuilder();
            lBuff.Append(mSubProt).Append('[').Append(mEncoding.ToString()).Append("]");
            return lBuff.ToString();
        }

        public string GetSubProt()
        {
            return mSubProt;
        }

        public WebSocketEncoding GetEncoding()
        {
            return mEncoding;
        }

        public string GetNameSpace()
        {
            return mNameSpace;
        }

        public string GetFormat()
        {
            return mFormat;
        }
    }
}
